pgxn/neon/pagestore_fetchbuf.h

```diff
@@ -0,0 +1,79 @@
+#ifndef NEON_PAGESTORE_FETCHBUF_H
+#define NEON_PAGESTORE_FETCHBUF_H
+
+#include "storage/buf_internals.h"
+
+/*
+ * Prefetch implementation:
+ * 
+ * Prefetch is performed locally by each backend.
+ * 
+ * 
+ * There can be up to MAX_PREFETCH_REQUESTS registered using smgr_prefetch
+ * before smgr_read. All this requests are appended to primary smgr_read request.
+ * It is assumed that pages will be requested in prefetch order.
+ * Reading of prefetch responses is delayed until them are actually needed (smgr_read).
+ * It make it possible to parallelize processing and receiving of prefetched pages.
+ * In case of prefetch miss or any other SMGR request other than smgr_read,
+ * all prefetch responses has to be consumed.
+ */
+
+/* Max amount of tracked buffer reads */
+#define READ_BUFFER_SIZE 128
+
+typedef enum PrefetchStatus {
+	PRFS_EMPTY = 0,
+	PRFS_REQUESTED,
+	PRFS_RECEIVED,
+	PRFS_TAG_REMAINS,
+} PrefetchStatus;
+
```

Comment:
A comment to explain what each value means would be nice. All the other ones are quite self-explanatory actually, but I cannot guess what PRFS_TAG_REMAINS means (haven't read the rest of the patch yet).